core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/AbstractCPropertyTab.java

```diff
@@ -496,28 +496,37 @@ public static String getVariableDialog(Shell shell, ICConfigurationDescription c
 	}
 
 	public static String getWorkspaceDirDialog(Shell shell, String text) {
-		return getWorkspaceDialog(shell, text, true, null);
+		return getWorkspaceDialog(shell, text, true, true, null);
 	}
 
 	public static String getWorkspaceFileDialog(Shell shell, String text) {
-		return getWorkspaceDialog(shell, text, false, null);
+		return getWorkspaceDialog(shell, text, false, true, null);
+	}
+
+	public static String getWorkspaceDirDialog(Shell shell, String text, IProject prj) {
```

Comment:
Needs `@since` - but we can resolve that later in the review process

Reply:
Which version number is to be used?

Reply:
The minor version in MANIFEST.MF needs to be increased and use that number. The auto-fix in Eclipse should do that if you have API baseline setup. If you don't have it setup I think it will be `8.1`

I can handle that as part of the merging if it isn't straightforward for you.